uaclient/cli.py

```diff
@@ -293,13 +294,13 @@ def action_attach(args, cfg):
         return 0
     if os.getuid() != 0:
         raise exceptions.NonRootUserError()
-    contract_token = args.token
-    if not contract_token:
-        print("No valid contract token available")
-        return 1
+    if not args.token:
+        raise exceptions.UserFacingError(
```

Comment:
What do you think to converting this (and the other introduced `raise`s) to their own exception classes with the message hard-coded?  (So this would be `raise exception.AttachRequiresToken()` or something along those lines.)

Reply:
@OddBloke I had thought about this, but avoided it only because we didn't do any specific handling of that unique error condition. If we have 5 different exceptions defined and we only print(str(e)) and exit(1) I'd like to avoid having the unique exceptions defined as it is just more code.   I would lean slightly toward only defining separate exceptions if we have unique handling behavior associated with processing them. But, it's only a weak suggestion. I can be convinced otherwise.